src/promptflow/promptflow/executor/_tool_invoker.py

```diff
@@ -2,9 +2,83 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
 
+from dataclasses import dataclass
+from functools import partial
+from typing import Callable, Dict
+
 from promptflow._core.tool import ToolInvoker
+from promptflow.contracts.flow import InputAssignment, Node, ToolSource
+from promptflow.executor._docstring_parser import DocstringParser
+from promptflow.executor._tool_resolver import ToolResolver
 
 
 class DefaultToolInvoker(ToolInvoker):
     def invoke_tool(self, f, *args, **kwargs):
         return f(*args, **kwargs)  # Do nothing
+
+
+@dataclass
+class AssistantTool:
+    name: str
+    definition: dict
+    func: Callable
+
+
+class AssistantToolInvoker:
+    def __init__(self):
+        self._assistant_tools: Dict[str, AssistantTool] = {}
+
+    def load_tools(self, tools: list):
+        tool_resolver = ToolResolver.active_instance()
+        for tool in tools:
+            if tool["type"] != "function":
```

Comment:
Not critical. 

But personally, I may prefer to have a clear logic here:

```python
if tool["type"] in ("code_interpreter", "retrieval"):
    func_name = tool["type"]
    .....
else if tool["type"] == "function":
   .....
else:
    print(f"unknown tool type: {tool["type"]}")
```

Reply:
Refined.